src/adjoint.jl

```diff
@@ -30,7 +30,14 @@ function adjoint end
 function _pullback end
 function pullback end
 
-function gradm(ex, mut = false)
+
+function unthunk_tangent end
+@inline unthunk_tangent(x) = x
```

Comment:
Shouldn't this unthunk?

Reply:
FluxML/Zygote.jl#966 adds

```julia
@inline unthunk_tangent(x::AbstractThunk) = unthunk(x)
```

I didn't want to do it here, to avoid a dependency on ChainRulesCore.

Reply:
I suggest just adding it here.
A dependency on ChainRulesCore adds 0.05 seconds to load time,
and it has hit 1.0 so it is stable.

Reply:
Sure, why not - people should depend more on ChainRulesCore than ZygoteRules anyway in the future.

Ok with you, @DhairyaLGandhi ?

Reply:
I"m thinking back to whether we should be aggressive about `unthunking` in Zygote. Is the idea to then be able to thunk and unthunk inside `@adjoint`s?

Reply:
The idea is that Zygote.`@adjoint` will unthunk automatically, to preserve backwards-compatibility, but thunks will be passed through to `ChainRulesCore.rrule`, since ChainRulesCore v1.0, `rrule`s are required to accept chunks.

That way, nothing changes for existing `@adjoint`s, but `rrule`s will only unthunk if necessary, with potentially very significant savings in computation time (see benchmarks in FluxML/Zygote.jl#966).

CC @oxinabox